docs/eip/5559.md

```diff
@@ -0,0 +1,447 @@
+
+3. Each `StorageHandledBy__()` proposal must define the precise formatting of any message payloads that require signatures and complete descriptions of custom cryptographic techniques implemented for additional security, accessibility or privacy.
+
+## Implementation featuring ENS on L2 & Database
+ENS off-chain resolvers capable of reading from and writing to databases are perhaps the most common use-case for CCIP-Read and CCIP-Write. One example of such a (minimal) resolver is given below along with the client-side code for handling the storage handler revert.
+
+### L1 Contract
+```solidity
+/* ENS resolver implementing StorageHandledByDatabase() */
+interface iResolver {
+    // Defined in EIP-5559
+    error StorageHandledByL2(
+        uint chainId,
+        address contractL2
+    );
+    error StorageHandledByDatabase(
+        string gatewayUrl
+    );
+    // Defined in EIP-137
+    function setAddr(bytes32 node, address addr) external;
+}
+
+// Defined in EIP-5559
+string public gatewayUrl = "https://post.namesys.xyz"; // RESTful API endpoint
+uint256 public chainId = uint(21); // ChainID of L2
+address public contractL2 = "0x839B3B540A9572448FD1B2335e0EB09Ac1A02885"; // Contract on L2
+
+/**
+* Sets the ethereum address associated with an ENS node
+* [!] May only be called by the owner or manager of that node in ENS registry
+* @param node Namehash of ENS domain to update
+* @param addr Ethereum address to set
+*/
+function setAddr(
+    bytes32 node,
+    address addr
+) authorised(node) {
+    // Defer to database storage
+    revert StorageHandledByDatabase(
+        gatewayUrl
+    );
+}
+
+/**
+* Sets the avatar text record associated with an ENS node
+* [!] May only be called by the owner or manager of that node in ENS registry
+* @param node Namehash of ENS domain to update
+* @param key Key for ENS text record
+* @param value URL to avatar
+*/
+function setText(
+    bytes32 node,
+    string key,
+    string value
+) external {
+    // Verify owner or manager permissions
+    require(authorised(node), "NOT_ALLOWED");
+    // Defer to L2 storage
+    revert StorageHandledByL2(
+        chainId, 
+        contractL2
+    );
+}
+```
+
+### L2 Contract
+```solidity
+// Function in L2 contract
+function setText(
+    bytes32 node,
+    bytes32 key,
+    bytes32 value
+) external {
+    // Store record mapped by node & sender
+    records[keccak256(abi.encodePacked(node, msg.sender))]["text"][key] = value;
+}
+```
+
+### Client-side Code
+```ts
+/* Client-side pseudo-code in ENS App */
+// Deterministically generate signer keypair
+let signer = keygen(username, sigKeygen, password);
+// Construct POST body by signing calldata with derived private key
+let post: Post = signData(node, addr, signer.priv);
+// POST to gateway
+await fetch(gatewayUrl, {
+  method: "POST",
+  body: JSON.stringify(post)
+});
+```
+
+## Backwards Compatibility
+Methods in this document are not compatible with previous EIP-5559 specifications.
```

Comment:
This is incorrect and needs fixing. L2 handler is unchanged

Reply:
Fixed by https://github.com/ensdomains/docs/pull/224/commits/f50c32183c4aae26c06e7107f4ae8be9ff918601